docs/devguide.rst

```diff
@@ -61,22 +61,66 @@ is open to interpretation.
   unittest directly or write tests in an object-oriented style.
 * Test generators make more dots and the dots must flow!
 
+You can easily check for style issues, including some outright bugs such
+as mispelled variable names, using pylint. If you're using Anaconda you'll
+need to run "conda install pylint" once. You can easily run pylint on
+the edited files in your uncommited git change:
+
+    $ make lint
+
+If you want to lint the entire code base run:
+
+    $ make lint-all
+
 How to Test
 ================
-First, install nose: http://nose.readthedocs.org/en/latest/
+First, install nose: http://nose.readthedocs.org/en/latest/. Second, ensure
+your cwd is the root directory of the project (i.e., the one containing the
+.git directory).
 
 To perform all unit tests::
 
-    $ cd tests/
-    $ nosetests
+    $ make test-all
```

Comment:
> you're not actually testing the code in the ../xonsh directory; 

Right. I personally don't want to test the code that is in the ../xonsh directory.  I want to test the version of xonsh that is installed locally.  I understand that this is not what everyone wants and some people do want to test what is in the ../xonsh dir. I am happy to support that workflow and encourage people to use whatever workflow works best for them.